File: test_data/code/python/1582-special-positions-in-a-binary-matrix.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        m, n = len(mat), len(mat[0])
        res = 0
        for i in range(m):
            yes = []
            for j in range(n):
                if mat[i][j] == 1:
                    yes.append([i, j])
                    for k in range(m):
                        if k != i and mat[k][j] == 1:
                            break
                    for k in range(n):
                        if k != j and mat[i][k] == 1:
                            break
                    break
            for item in yes:
                res += 1
        return res

File: test_data/code/java/1582-special-positions-in-a-binary-matrix.java
class Solution {
    public int numSpecial(int[][] mat) {
        int count = 0;
        for (int i = 0; i < mat.length; i++) {
            boolean rowHasOne = false;
            for (int j = 0; j < mat[i].length; j++) {
                if (mat[i][j] == 1) {
                    rowHasOne = true;
                }
            }
            if (rowHasOne) {
                boolean colHasOne = false;
                for (int j = 0; j < mat[i].length; j++) {
                    if (mat[j][i] == 1 && j != i) {
                        colHasOne = true;
                        break;
                    }
                }
                if (!colHasOne) {
                    count++;
                }
            }
        }
        return count;
    }
}